src/libvirtApi/snapshot.js

```diff
@@ -30,9 +30,27 @@ import { call, Enum, timeout } from './helpers.js';
 import { logDebug } from '../helpers.js';
 
 export function snapshotCreate({ connectionName, vmId, name, description, memoryPath, disks, isExternal, storagePools }) {
-    // that flag ought to be implicit for non-running VMs, see https://issues.redhat.com/browse/RHEL-22797
+    // The "disk only" flag ought to be implicit for non-running VMs, see https://issues.redhat.com/browse/RHEL-22797
+
+    // However, "disk only" can be used to request external snapshots
+    // for a stopped machine. The alternative is to list all disks
+    // with a snapshot type of "external" in the XML, but then we need
+    // to worry about which disks to include and which to skip.
+
+    // The behavior is as follows:
+    //
+    // VM state  |  disk only  |  memory path   =>  resulting snapshot type
+    // --------------------------------------------------------------------
+    // shutoff   |  false      |  null          =>  internal
+    // shutoff   |  true       |  null          =>  external
+    // running   |  false      |  null          =>  internal full system
+    // running   |  false      |  non-null      =>  external full system
+    // running   |  true       |  null          =>  external disk only
```

Comment:
That's the case which I find crazy. Can we add some assertion that we never ever attempt this?

Reply:
Thanks!